lib/opentelemetry/trace/batch_span_processor.lua

```diff
@@ -125,71 +76,75 @@ function _M.on_end(self, span)
         return
     end
 
+    -- Drop span if queue is full, otherwise add span to queue
     if self:get_queue_size() >= self.max_queue_size then
-        -- drop span
         if self.drop_on_queue_full then
             ngx.log(ngx.WARN, "queue is full, drop span: trace_id = ", span.ctx.trace_id, " span_id = ", span.ctx.span_id)
             self.dropping_count = self.dropping_count + 1
-            return
         end
-
-        -- export spans
-        process_batches_timer(self, self.batch_to_process)
-        self.batch_to_process = {}
-    end
-
-    table.insert(self.queue, span)
-    if #self.queue == 1 then
-        self.first_queue_t = now()
```

Comment:
if `self:get_queue_size() >= self.max_queue_size` is true, and `self.drop_on_queue_full = false`, the span is lost.